editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/preferences/PreferredLabelDialog.java

```diff
@@ -0,0 +1,210 @@
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.fusesource.ide.camel.editor.internal.UIMessages;
+import org.fusesource.ide.camel.model.service.core.catalog.CamelModel;
+import org.fusesource.ide.camel.model.service.core.catalog.CamelModelFactory;
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+
+/**
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class PreferredLabelDialog extends TitleAreaDialog {
+
+	private String component;
+	private String parameter;
+
+	private Combo componentCombo;
+	private Combo parameterCombo;
+
+	private IInputValidator componentValidator;
+	private IInputValidator parameterValidator;
+
+	public PreferredLabelDialog(Shell parentShell) {
+		super(parentShell);
+	}
+
+	public String getComponent() {
+		return component;
+	}
+
+	public void setComponent(String component) {
+		this.component = component;
+	}
+
+	public String getParameter() {
+		return parameter;
+	}
+
+	public void setParameter(String parameter) {
+		this.parameter = parameter;
+	}
+
+	public void setComponentValidator(IInputValidator componentValidator) {
+		this.componentValidator = componentValidator;
+	}
+
+	public void setParameterValidator(IInputValidator parameterValidator) {
+		this.parameterValidator = parameterValidator;
+	}
+
+	public void validate() {
+		if (componentValidator != null) {
+			if (setWarning(componentValidator.isValid(componentCombo.getText()))) {
+				return;
+			}
+		}
+		if (parameterValidator != null) {
+			setWarning(parameterValidator.isValid(parameterCombo.getText()));
+		}
+	}
+
+	public boolean setWarning(String msg) {
+		if ((msg == null || msg.isEmpty()) && getButton(IDialogConstants.OK_ID) != null) {
+			getButton(IDialogConstants.OK_ID).setEnabled(true);
+			setMessage(null, IMessageProvider.NONE);
+			return false;
+		}
+		setMessage(msg, IMessageProvider.WARNING);
+		getButton(IDialogConstants.OK_ID).setEnabled(false);
+		return true;
+	}
+
+	@Override
+	public void create() {
+		super.create();
+		setTitle(UIMessages.preferredLabels_title);
+		validate();
+	}
+
+	@Override
+	protected Control createDialogArea(Composite parent) {
+		Composite area = (Composite) super.createDialogArea(parent);
+		Composite container = new Composite(area, SWT.NONE);
+		container.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		GridLayout layout = new GridLayout(2, false);
+		container.setLayout(layout);
+
+		createComponentComboBox(container);
+		createParameterCombo(container);
+
+		CamelModel model = CamelModelFactory.getModelForVersion(CamelModelFactory.getLatestCamelVersion());
+		List<String> componentNames = new ArrayList<String>();
+		model.getEipModel().getSupportedEIPs().stream().map(eip -> eip.getName()).sorted().forEach(componentNames::add);
+
+		componentCombo.setItems(componentNames.toArray(new String[componentNames.size()]));
```

Comment:
you should better use a ComboViewer.
As input, you will provide the Eip.
It will be easier to manipulate then, for instance l.177, no more need to do a model.getEipModel().getEIPByName(componentName).getParameters() you will be able to directly ge the parameters.
Between the componentCombo and the parameterCombo, we can use a Master-Detail pattern for Databinding.